lib/src/database.dart

```diff
@@ -152,6 +164,47 @@ class Query {
         .then((snapshot) => new DataSnapshot(snapshot));
   }
 
+  /// Detaches a callback previously attached with [on].
+  ///
+  /// Note that if [on] was called multiple times with the same [eventType] and
+  /// [callback], the callback will be called multiple times for each event, and
+  /// [off] must be called multiple times to remove the callback. Calling [off]
+  /// on a parent listener will not automatically remove listeners registered on
+  /// child nodes, [off] must also be called on any child listeners to remove
+  /// the callback.
```

Comment:
This paragraph is now misleading after all recent changes as there is no `callback` argument. We need to re-phrase it or even move these details to `QuerySubscription.cancel` docs.